Feature/change slider renderer to spinner

sluke-nuix opened the pull request:
Change the JSlider control on the CustomTabPanel to use a JSpinner for numeric display so the value can be modified manually.

Also removed the .pom file.  It was getting in the way and I am not sure of the need for it.

sluke-nuix committed:
Add a second level of ANDed filters to dynamic tables

sluke-nuix committed:
Merge branch 'feature/toast' into feature/dynamictable_and_filtering

sluke-nuix committed:
Expanded visibility for filtering on dynamic table to allow editing them from external source

sluke-nuix committed:
Merge branch 'master' into feature/dynamictable_and_filtering

sluke-nuix committed:
Move images to resources directory

This helps making the images available for builds with fewer custom steps.

sluke-nuix committed:
Add spinner to JSlider control

sluke-nuix committed:
Merge branch 'feature/dynamictable_and_filtering' into feature/change_slider_renderer_to_spinner

sluke-nuix committed:
Ensure JSlider updates inform the linked JSpinner

sluke-nuix commented:
This is a branch off of a banch that did some additional filtering for dynamic tables.

JuicyDragon commented:
So I opened the project in Eclipse, built the JAR and attempted to run the example from the repository `Nx\Examples\TabbedCustomDialog.rb`.  It appears that (at least for me) the image resources are no longer being included in the JAR

```
java.lang.NullPointerException
	at sun.awt.SunToolkit.getImageFromHash(sun/awt/SunToolkit.java:692)
	at sun.awt.SunToolkit.getImage(sun/awt/SunToolkit.java:728)
	at com.nuix.nx.dialogs.TabbedCustomDialog.<init>(com/nuix/nx/dialogs/TabbedCustomDialog.java:114)
	at com.nuix.nx.dialogs.TabbedCustomDialog.<init>(com/nuix/nx/dialogs/TabbedCustomDialog.java:104)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(jdk/internal/reflect/NativeConstructorAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(jdk/internal/reflect/DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(java/lang/reflect/Constructor.java:490)
	at org.jruby.javasupport.JavaConstructor.newInstanceDirect(org/jruby/javasupport/JavaConstructor.java:253)
	at org.jruby.RubyClass.new(org/jruby/RubyClass.java:881)
	at org.jruby.RubyClass$INVOKER$i$newInstance.call(org/jruby/RubyClass$INVOKER$i$newInstance.gen)
	at RUBY.<main>(C:\Work\GitHub\Nx\Examples\TabbedCustomDialog.rb:16)
	at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:1292)
	at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:1296)
	at org.jruby.embed.internal.EmbedEvalUnitImpl.run(org/jruby/embed/internal/EmbedEvalUnitImpl.java:119)
	at org.jruby.embed.jsr223.JRubyEngine.doEval(org/jruby/embed/jsr223/JRubyEngine.java:97)
	at org.jruby.embed.jsr223.JRubyEngine.eval(org/jruby/embed/jsr223/JRubyEngine.java:124)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(jdk/internal/reflect/NativeMethodAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(jdk/internal/reflect/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:566)
	at com.nuix.script.e.a(com/nuix/script/SourceFile:53)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.nuix.script.e.a(com/nuix/script/SourceFile:50)
	at com.sun.proxy.$Proxy44.eval(com/sun/proxy/$Proxy44)
	at com.nuix.script.a.a(com/nuix/script/SourceFile:272)
	at com.nuix.investigator.script.m.e(com/nuix/investigator/script/SourceFile:314)
	at com.nuix.investigator.script.m.d(com/nuix/investigator/script/SourceFile:299)
	at com.nuix.investigator.main.q.run(com/nuix/investigator/main/SourceFile:122)
	at com.nuix.investigator.main.b.run(com/nuix/investigator/main/SourceFile:112)
	at com.nuix.bootstrap.e$b.a(com/nuix/bootstrap/SourceFile:254)
	at com.nuix.bootstrap.e.run(com/nuix/bootstrap/SourceFile:115)
	at com.nuix.bootstrap.h.a(com/nuix/bootstrap/SourceFile:116)
	at com.nuix.investigator.main.DesktopConsoleMain.main(com/nuix/investigator/main/SourceFile:43)
```

Are you not seeing this?  I tinkered with it a bit and couldn't seem to get it right again.

**Side Note:** This library would likely benefit from being converted to Intellij/Gradle at some point.

sluke-nuix commented:
This started to happen to me in IntelliJ as well.   It was one of the reasons I removed the POM file.  I don't know what changed but the image files weren't moving to the build folder with the class files when it was built.  When I overrode the settings in IntelliJ, they would revert whenever I opened the IntelliJ and I would have to reset them.  So I removed the POM and switched to the IntelliJ builder system.


  1.  First I tried to do it the way you are "supposed" to do it - move all the items to a resources folder, and mark the folder as the Resources Root directory.
     *   This is supposed to make it so IntelliJ would find the files in there and automatically include them in the output
     *   This didn't work for me though so...
  2.  Second, I had to manually add the Resources folder into the archive:
[cid:259c8e61-fc0b-4638-9a94-eff6056a4fa3]

I imagine you will have to do the equivalent of that in Eclipse.  I wish I knew what was making the maven build not include the images (or any of the libraries during build) but I couldn't figure that out.




--

Steven Luke

Product Owner, SDK and Innovation

Conshohocken, PA


________________________________
From: Jason Wells ***@***.***>
Sent: Monday, June 12, 2023 3:37 PM
To: Nuix/Nx ***@***.***>
Cc: Steven Luke ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL]Re: [Nuix/Nx] Feature/change slider renderer to spinner (PR #19)


So I opened the project in Eclipse, built the JAR and attempted to run the example from the repository Nx\Examples\TabbedCustomDialog.rb. It appears that (at least for me) the image resources are no longer being included in the JAR

java.lang.NullPointerException
        at sun.awt.SunToolkit.getImageFromHash(sun/awt/SunToolkit.java:692)
        at sun.awt.SunToolkit.getImage(sun/awt/SunToolkit.java:728)
        at com.nuix.nx.dialogs.TabbedCustomDialog.<init>(com/nuix/nx/dialogs/TabbedCustomDialog.java:114)
        at com.nuix.nx.dialogs.TabbedCustomDialog.<init>(com/nuix/nx/dialogs/TabbedCustomDialog.java:104)
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(jdk/internal/reflect/NativeConstructorAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(jdk/internal/reflect/DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(java/lang/reflect/Constructor.java:490)
        at org.jruby.javasupport.JavaConstructor.newInstanceDirect(org/jruby/javasupport/JavaConstructor.java:253)
        at org.jruby.RubyClass.new(org/jruby/RubyClass.java:881)
        at org.jruby.RubyClass$INVOKER$i$newInstance.call(org/jruby/RubyClass$INVOKER$i$newInstance.gen)
        at RUBY.<main>(C:\Work\GitHub\Nx\Examples\TabbedCustomDialog.rb:16)
        at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:1292)
        at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:1296)
        at org.jruby.embed.internal.EmbedEvalUnitImpl.run(org/jruby/embed/internal/EmbedEvalUnitImpl.java:119)
        at org.jruby.embed.jsr223.JRubyEngine.doEval(org/jruby/embed/jsr223/JRubyEngine.java:97)
        at org.jruby.embed.jsr223.JRubyEngine.eval(org/jruby/embed/jsr223/JRubyEngine.java:124)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(jdk/internal/reflect/NativeMethodAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(jdk/internal/reflect/DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:566)
        at com.nuix.script.e.a(com/nuix/script/SourceFile:53)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.nuix.script.e.a(com/nuix/script/SourceFile:50)
        at com.sun.proxy.$Proxy44.eval(com/sun/proxy/$Proxy44)
        at com.nuix.script.a.a(com/nuix/script/SourceFile:272)
        at com.nuix.investigator.script.m.e(com/nuix/investigator/script/SourceFile:314)
        at com.nuix.investigator.script.m.d(com/nuix/investigator/script/SourceFile:299)
        at com.nuix.investigator.main.q.run(com/nuix/investigator/main/SourceFile:122)
        at com.nuix.investigator.main.b.run(com/nuix/investigator/main/SourceFile:112)
        at com.nuix.bootstrap.e$b.a(com/nuix/bootstrap/SourceFile:254)
        at com.nuix.bootstrap.e.run(com/nuix/bootstrap/SourceFile:115)
        at com.nuix.bootstrap.h.a(com/nuix/bootstrap/SourceFile:116)
        at com.nuix.investigator.main.DesktopConsoleMain.main(com/nuix/investigator/main/SourceFile:43)


Are you not seeing this? I tinkered with it a bit and couldn't seem to get it right again.

Side Note: This library would likely benefit from being converted to Intellij/Gradle at some point.

—
Reply to this email directly, view it on GitHub<https://github.com/Nuix/Nx/pull/19#issuecomment-1587959959>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXHCBNTI32NO7ZMKATWIZCTXK5VXPANCNFSM6AAAAAAZDRDINM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>






CAUTION:This email originated from outside of Nuix. Do not click links or open attachments unless you recognise the sender and know the content is safe.

sluke-nuix committed:
Fixed label bug with spinner

JuicyDragon commented:
When I initially tried to review this a while back, I was having issues with the modifications.  I just converted the project to IntelliJ IDEA and Gradle on #20.  Once that looks good and gets merged in to master, we can reapply these changes.  Unfortunately will have to apply the changes manually since so much changed.  The good news is that going forward incorporating changes such as this should go much easier I think.

sluke-nuix committed:
Allow model to be passed into StringList
